Library/Homebrew/download_strategy.rb

```diff
@@ -518,10 +518,13 @@ def resolve_url_basename_time_file_size(url, timeout: nil)
       [*parse_content_disposition.call("Content-Disposition: #{header}")]
     end
 
-    time = parsed_headers
-           .flat_map { |headers| [*headers["last-modified"]] }
-           .map { |t| t.match?(/^\d+$/) ? Time.at(t.to_i) : Time.parse(t) }
-           .last
+    time =  parsed_headers
+            .flat_map { |headers| [*headers["last-modified"]] }
+            .filter_map do |t|
+              t.match?(/^\d+$/) ? Time.at(t.to_i) : Time.parse(t)
+            rescue ArgumentError
```

Comment:
What is it that's raising this: `match?`, `at` or `parse`? A comment might be nice explaining why this is being rescued, too.

Reply:
It's the parsing that's raising `ArgumentError`, because `Time.at` is only when the timestamp is all digits.

Reply:
Ok. Either a comment or restricting the code (with an `if`/`else`) to make that clearer would be great!